src/NuGet.Core/NuGet.DependencyResolver.Core/GraphItemKeyComparer.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.DependencyResolver
+{
+    public sealed class GraphItemKeyComparer<T> : IEqualityComparer<GraphItem<T>>
```

Comment:
Added.

Reply:
nit: Given that this is used *only* in NuGet.Commands, can we moved it there? I know the type is not defined there, but if there's only 1 usage, it might make sense to keep it there.

Reply:
I considered asking the same question (move comparer to NuGet.Commands and make it internal), but I guess it could be useful to someone, somewhere, somehow.

Reply:
> I guess it could be useful to someone, somewhere, somehow.

Only if they are implementing restore themselves :D

Reply:
I would prefer to let it unchanged in case that it may be needed.